### PR TITLE
Allow pandas 1.1 as dependency

### LINKED ISSUE #27: Setup requirement of pandas
The requirement of demandlib for pandas is [`pandas >= 0.18.0, < 0.25`](https://github.com/oemof/demandlib/blob/d10b09b9c31e3f87a49a66fc0d37b4c6954684ae/setup.py#L24).

I install the demandlib in a package that uses xarray, which is incompatible with demandlib as it requires pandas >= 0.25.  

Do you see a problem in loosening the limit and allowing pandas < 1.1.0 or `pandas >= 0.25.`? I didn't have a problem with that creating power and heat demand profiles but as there are no tests of the demandlib, I cannot be sure, of course.

### PR DESCRIPTION
**Pull request by p-snft**
Fix #27

### CODE REVIEW

**Comment by uvchik** [comment]
In the future pypi will raise an error if there are conflicting version numbers (s. below). Therefore I would suggest to remove the upper bound for the package version to avoid such problems.

```
ERROR: After October 2020 you may experience errors when installing or updating packages.
This is because pip will change the way that it resolves dependency conflicts.
```

**Comment by SabineHaas** [comment]
So shall we go for `'pandas >= 0.18.0'` , without upper limit?

**Comment by uvchik** [comment]
> So shall we go for `'pandas >= 0.18.0'` , without upper limit?

Yes, I would vote for this variant.

**Comment by uvchik** [comment]
Furthermore we could integrate Travis, so that we are noticed if a problem occurs.

File: setup.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def read(fname):
       long_description=read('README.rst'),
       packages=find_packages(),
       install_requires=['numpy >= 1.7.0, < 1.17',
-                        'pandas >= 0.18.0, < 0.25'],
+                        'pandas >= 0.18.0, < 1.2'],
```
Comment (SabineHaas) [inline setup.py:24]:
```suggestion
                        'pandas >= 0.18.0'],
```
already suggested ;)

**Review by SabineHaas — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (uvchik) [reply setup.py:24]: numpy, too?

Comment (SabineHaas) [reply setup.py:24]: But then we should check whether the main function work with the newest numpy versions, too..

**Comment by uvchik** [comment]
I would do the same for numpy. In #29 I added automatic test, so we will be informed if something goes wrong. And normally numpy and pandas inform people about API changes with future warnings.

**Review by uvchik — APPROVED** [review]
I would do the same for numpy. In #29 I added automatic test, so we will be informed if something goes wrong. And normally numpy and pandas inform people about API changes with future warnings.

**Comment by SabineHaas** [comment]
> I would do the same for numpy. In #29 I added automatic test, so we will be informed if something goes wrong. And normally numpy and pandas inform people about API changes with future warnings.

Nice! Thanks :+1:

**Pull request merged**